src/containers/Map/MapScreen.tsx

```diff
@@ -50,36 +54,95 @@ const MapScreen = () => {
 	const [showMapNodes, setShowMapNodes] = useState(false);
 
 	// Bot that was selected to send to some map node, used when showing map nodes
+	const botS: MarkerData | null = botSelected ? botSelected : null;
 	const [
 		selectedBotForOrder,
 		setSelectedBotForOrder,
-	] = useState<MarkerData | null>(null);
+	] = useState<MarkerData | null>(botS);
 
 	const [hasLocationPermission, setLocationPermission] = useState('null');
 	const [alert, setAlert] = useState(false);
 
 	const [loading, setLoading] = useState<boolean>(false);
 	const { state } = useContext(Ctx);
 
+	async function addToRoute(marker: MarkerData) {
+		console.log('adding marker ' + marker.name);
+		let curRoute = botRoute ? botRoute : [];
+		if (curRoute.indexOf(marker) > -1) {
+			await removeFromRoute(marker);
+		} else {
+			if (curRoute.length === 0 && selectedBotForOrder) {
+				let curPaths = paths ? paths : [];
```

Comment:
Tiny nitpick, for this common ternary pattern that shows up a couple times, you can also use `let curPaths = paths || [];` (if you want 😄 )

In JS `||, &&` act as selection operators in addition to boolean ones.